File: scanner.js
const puppeteer = require('puppeteer-core');

async function getDomContent(url, selector) {
    const browser = await puppeteer.launch({
        executablePath: '/Applications/Google Chrome.app/Contents/MacOS/Google Chrome',
        headless: true 
        });
    const page = await browser.newPage();
    await page.goto(url, { waitUntil: 'networkidle0', timeout: 450000 });

    const data = await page.$$eval(selector, (elements) => {
        return elements.map(el => ({
            text: el.innerText.trim(),
            html: el.innerHTML,
            tag: el.tagName,
            attributes: Array.from(el.attributes).reduce((acc, attr) => {
                acc[attr.name] = attr.value;
                return acc;

            }, {}),
        }))
    })

    await browser.close();
    return data;

}



getDomContent('https://docs.browser-use.com/quickstart', 'h2').then(console.log);

File: learning/DOM/scanner.py
import asyncio
from playwright.async_api import async_playwright

async def get_dom_content(url, selector):
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        page = await browser.new_page()
        await page.goto(url, wait_until='networkidle')

        await page.wait_for_selector(selector)

        elements = await page.query_selector_all(selector)

        results = []
        for el in elements:
            results.append({
                "text": await el.inner_text(),
                "svg": await el.get_attribute("svg"),
            })
        await browser.close()
        return results